docs/learn.md

```diff
@@ -126,10 +126,281 @@ OCaml. Some examples are the upgrade of the compiler version used by Melange to
+project or use the `opam switch` command to identify if a switch already exists
+in the project folder.
+
+If it does not exist, we can create it with:
+
+```bash
+opam switch create . 4.14.1 --deps-only
+```
+
+If it exists, we can install the dependencies of the project with:
+
+```bash
+opam install . --deps-only
+```
+
+#### Add new packages
+
+To add a new package to the opam switch, we can do:
+
+```bash
+opam install <package_name>
+```
+
+But opam will not modify the `app.opam` file during the installation, this has
+to be done by hand. It is as simple as adding the name of the package in the
+`depends` field.
+
+#### Linking packages for development
+
+This can be achieved with `opam pin`. For example, to pin a package to a
+specific commit on GitHub:
+
+```bash
+opam pin add reason-react.dev https://github.com/reasonml/reason-react.git#61bfbfaf8c971dec5152bce7e528d30552c70bc5
+```
+
+Branch names can also be used.
+
+```bash
+opam pin add reason-react.dev https://github.com/reasonml/reason-react.git#feature
+```
+
+A shortcut to get the latest version of a package is to use the `--dev-repo`
+flag, e.g.
+
+```bash
+opam pin add reason-react.dev --dev-repo
+```
+
+To remove the pinning for any package, use `opam unpin <package_name>` or `opam
+pin remove <package_name>`.
+
+For other options, the command is well described in [the official
+documentation](https://opam.ocaml.org/doc/Usage.html#opam-pin).
+
+#### Upgrading packages
+
+To upgrade the installed packages to the latest version, run:
+
+```bash
+opam upgrade <package_name>
+```
+
+`opam upgrade` is also able to upgrade *all* the packages of the local switch if
+no package name is given.
+
+There is one big difference compared to npm: opam stores a local copy of the
+opam repository, like `apt-get` does in Debian. So we often want to update this
+copy before requesting an upgrade:
+
+```bash
+opam update && opam upgrade <package_name>
+```
+
+#### Dev dependencies
+
+You can use the [`with-dev-setup`
+field](https://opam.ocaml.org/doc/Manual.html#pkgvar-with-dev-setup) to define
+dependencies that are only required at development time. For example:
+
+```
+depends: [
+  "ocamlformat" {with-dev-setup}
+]
+```
+
+This has to be combined with the `--with-dev-setup` flag when installing
+dependencies, e.g. `opam install --deps-only --with-dev-setup`.
+
+#### Lock files
+
+Lock files are also not common yet in the opam world, but they can be used as
+follows:
+
+- Using `opam lock` to generate the lock file when needed (basically after each
+  `opam install` or `opam upgrade`).
+- Adding `--locked` to all the `opam install --deps-only` and `opam switch
+  create .` commands.
+
+
+#### Bindings and package management
+
+When writing Melange libraries that bind to existing JavaScript packages, the
+users of the Melange library will have to make sure to install those JavaScript
+packages manually.
+
+This is similar to how OCaml bindings to system libraries work, see examples
```

Comment:
mention "depext"

Reply:
I don't think it adds anything but confusion, as "depext" is not compatible with Melange / JavaScript packages.